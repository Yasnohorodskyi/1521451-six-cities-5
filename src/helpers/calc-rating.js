import PropTypes from 'prop-types';

const calcRating = (percent) => {
  return {
    width: percent * 10 * 2 + `%`
  };
};

calcRating.propTypes = {
  percent: PropTypes.number.isRequired,
};

export {calcRating};
