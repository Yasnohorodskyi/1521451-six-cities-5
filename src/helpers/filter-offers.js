const data = {
  index: 0
};
const filterOffers = (currentOffer, maxElements, id) => {
  if (!maxElements) {
    return `main`;
  } else if (maxElements && data.index < maxElements) {
    if (Number(id) !== Number(currentOffer)) {
      data.index++;
      return `other`;
    }
  }
  return null;
};

export {filterOffers};
