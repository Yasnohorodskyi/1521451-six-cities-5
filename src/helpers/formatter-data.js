import PropTypes from 'prop-types';


const formatYearMonthDay = (time) => {

  const timestamp = new Date(time).getTime();

  let options = {
    year: `numeric`,
    month: `numeric`,
    day: `numeric`
  };

  const formatter = new Intl.DateTimeFormat(`en-EN`, options).format(timestamp);
  return formatter.toString().replace(/\//g, `-`);

};

const formatMonthYear = (time) => {
  const options = {month: `long`, year: `numeric`};
  const date = new Date(time);
  return new Intl.DateTimeFormat(`en-EN`, options).format(date);
};

const formatterData = (time, type) => {
  switch (type) {
    case `MonthDay`:
      return formatMonthYear(time);
    case `YearMonthDay`:
      return formatYearMonthDay(time);
    default:
      return time;
  }
};

formatterData.propTypes = {
  time: PropTypes.number.isRequired,
  type: PropTypes.string.isRequired,
};

export {formatterData};
