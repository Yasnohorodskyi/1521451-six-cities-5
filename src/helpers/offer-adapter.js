export const offerAdapter = (data) => {

  const newOffer = JSON.parse(JSON.stringify(data));
  newOffer.isFavorite = data.isFavorite ? data.isFavorite : data.is_favorite;
  return newOffer;
  /*
  const newOffer = {
    ...data,
    isFavorite: data.isFavorite ? data.isFavorite : data.is_favorite
  };

  return newOffer;
  */
};
