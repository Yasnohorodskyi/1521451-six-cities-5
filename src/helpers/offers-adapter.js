import {extend} from './extend';

export const offersAdapter = (offers) => {
  let newOfferArray = [];
  offers.forEach(function (offer) {
    let offerObject = extend(offer, {
      isFavorite: offer.is_favorite,
      isPremium: offer.is_premium,
      maxAdults: offer.max_adults,
      previewImage: offer.preview_image
    });
    newOfferArray.push(offerObject);
    delete offerObject[`is_favorite`];
    delete offerObject[`is_premium`];
    delete offerObject[`max_adults`];

  });

  return newOfferArray;
};

export const offerAdapter = (offer) => {

  let offerObject = extend(offer, {
    isFavorite: offer.is_favorite,
    isPremium: offer.is_premium,
    maxAdults: offer.max_adults,
    previewImage: offer.preview_image
  });

  delete offerObject[`is_favorite`];
  delete offerObject[`is_premium`];
  delete offerObject[`max_adults`];
  delete offerObject[`preview_image`];

  return offerObject;
};
