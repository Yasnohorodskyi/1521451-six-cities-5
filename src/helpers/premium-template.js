import React from 'react';
import PropTypes from 'prop-types';

const premiumTemplate = (className) => {
  return (
    <div className={className}>
      <span>Premium</span>
    </div>
  );
};

premiumTemplate.propTypes = {
  className: PropTypes.string.isRequired,
};

export {premiumTemplate};
