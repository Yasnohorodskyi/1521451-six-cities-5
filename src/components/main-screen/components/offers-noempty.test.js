import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import OffersNoempty from "./offers-noempty";
import {offers, cities} from "../../../mocks-for-tests/mocks";
import {offersAdapter} from "../../../helpers/offers-adapter";

it(`OffersNoempty is rendered correctly`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <OffersNoempty
        offers={offersAdapter(offers)}
        currentCity={cities[`Brussels`]}
      />
  );
  expect(tree).toMatchSnapshot();
});
