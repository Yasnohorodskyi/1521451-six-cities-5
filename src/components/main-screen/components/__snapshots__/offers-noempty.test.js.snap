// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersNoempty is rendered correctly 1`] = `
<div
  className="cities"
>
  <div
    className="cities__places-container container"
  >
    <section
      className="cities__places places"
    >
      <h2
        className="visually-hidden"
      >
        Places
      </h2>
      <b
        className="places__found"
      >
         
        4
         places to stay in  
        Brussels
      </b>
      <OnClickOutside(WithOpenSelect)
        currentCity={
          Object {
            "location": Object {
              "latitude": 50.846558,
              "longitude": 4.351697,
              "zoom": 13,
            },
            "name": "Brussels",
          }
        }
        eventTypes={
          Array [
            "mousedown",
            "touchstart",
          ]
        }
        excludeScrollbar={false}
        offers={
          Array [
            Object {
              "bedrooms": 5,
              "city": Object {
                "location": Object {
                  "latitude": 53.550343,
                  "longitude": 10.000654,
                  "zoom": 13,
                },
                "name": "Hamburg",
              },
              "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
              "goods": Array [
                "Washer",
                "Breakfast",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 11,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 53.53434,
                "longitude": 9.998654,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 190,
              "rating": 4.3,
              "title": "The house among olive ",
              "type": "house",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 48.85661,
                  "longitude": 2.351499,
                  "zoom": 13,
                },
                "name": "Paris",
              },
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "goods": Array [
                "Air conditioning",
                "Baby seat",
                "Towels",
                "Washer",
                "Breakfast",
                "Fridge",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 87,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 48.86561,
                "longitude": 2.350499,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 266,
              "rating": 2.9,
              "title": "Tile House",
              "type": "hotel",
            },
            Object {
              "bedrooms": 4,
              "city": Object {
                "location": Object {
                  "latitude": 50.846558,
                  "longitude": 4.351697,
                  "zoom": 13,
                },
                "name": "Brussels",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "goods": Array [
                "Washer",
                "Laptop friendly workspace",
                "Breakfast",
                "Air conditioning",
                "Baby seat",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 50.86556,
                "longitude": 4.371697,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 218,
              "rating": 4.6,
              "title": "The house among olive ",
              "type": "apartment",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 50.938362,
                  "longitude": 6.959974,
                  "zoom": 13,
                },
                "name": "Cologne",
              },
              "description": "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
              "goods": Array [
                "Breakfast",
                "Washer",
                "Baby seat",
                "Laptop friendly workspace",
                "Air conditioning",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 34,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 50.945362,
                "longitude": 6.935974,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 485,
              "rating": 2.5,
              "title": "Nice, cozy, warm big bed apartment",
              "type": "house",
            },
          ]
        }
        outsideClickIgnoreClass="ignore-react-onclickoutside"
        preventDefault={false}
        stopPropagation={false}
      />
      <div
        className="cities__places-list places__list tabs__content"
      >
        <Memo(Connect(OfferContainer))
          currentCity={
            Object {
              "location": Object {
                "latitude": 50.846558,
                "longitude": 4.351697,
                "zoom": 13,
              },
              "name": "Brussels",
            }
          }
          offers={
            Array [
              Object {
                "bedrooms": 5,
                "city": Object {
                  "location": Object {
                    "latitude": 53.550343,
                    "longitude": 10.000654,
                    "zoom": 13,
                  },
                  "name": "Hamburg",
                },
                "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                "goods": Array [
                  "Washer",
                  "Breakfast",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 11,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                ],
                "isFavorite": undefined,
                "isPremium": undefined,
                "location": Object {
                  "latitude": 53.53434,
                  "longitude": 9.998654,
                  "zoom": 16,
                },
                "maxAdults": undefined,
                "previewImage": undefined,
                "price": 190,
                "rating": 4.3,
                "title": "The house among olive ",
                "type": "house",
              },
              Object {
                "bedrooms": 2,
                "city": Object {
                  "location": Object {
                    "latitude": 48.85661,
                    "longitude": 2.351499,
                    "zoom": 13,
                  },
                  "name": "Paris",
                },
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "goods": Array [
                  "Air conditioning",
                  "Baby seat",
                  "Towels",
                  "Washer",
                  "Breakfast",
                  "Fridge",
                  "Laptop friendly workspace",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 87,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                ],
                "isFavorite": undefined,
                "isPremium": undefined,
                "location": Object {
                  "latitude": 48.86561,
                  "longitude": 2.350499,
                  "zoom": 16,
                },
                "maxAdults": undefined,
                "previewImage": undefined,
                "price": 266,
                "rating": 2.9,
                "title": "Tile House",
                "type": "hotel",
              },
              Object {
                "bedrooms": 4,
                "city": Object {
                  "location": Object {
                    "latitude": 50.846558,
                    "longitude": 4.351697,
                    "zoom": 13,
                  },
                  "name": "Brussels",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "goods": Array [
                  "Washer",
                  "Laptop friendly workspace",
                  "Breakfast",
                  "Air conditioning",
                  "Baby seat",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 3,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                ],
                "isFavorite": undefined,
                "isPremium": undefined,
                "location": Object {
                  "latitude": 50.86556,
                  "longitude": 4.371697,
                  "zoom": 16,
                },
                "maxAdults": undefined,
                "previewImage": undefined,
                "price": 218,
                "rating": 4.6,
                "title": "The house among olive ",
                "type": "apartment",
              },
              Object {
                "bedrooms": 2,
                "city": Object {
                  "location": Object {
                    "latitude": 50.938362,
                    "longitude": 6.959974,
                    "zoom": 13,
                  },
                  "name": "Cologne",
                },
                "description": "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
                "goods": Array [
                  "Breakfast",
                  "Washer",
                  "Baby seat",
                  "Laptop friendly workspace",
                  "Air conditioning",
                ],
                "host": Object {
                  "avatarUrl": "img/avatar-angelina.jpg",
                  "id": 25,
                  "isPro": true,
                  "name": "Angelina",
                },
                "id": 34,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                ],
                "isFavorite": undefined,
                "isPremium": undefined,
                "location": Object {
                  "latitude": 50.945362,
                  "longitude": 6.935974,
                  "zoom": 16,
                },
                "maxAdults": undefined,
                "previewImage": undefined,
                "price": 485,
                "rating": 2.5,
                "title": "Nice, cozy, warm big bed apartment",
                "type": "house",
              },
            ]
          }
        />
      </div>
    </section>
    <div
      className="cities__right-section"
    >
      <Memo(Connect(MapContainer))
        offers={
          Array [
            Object {
              "bedrooms": 5,
              "city": Object {
                "location": Object {
                  "latitude": 53.550343,
                  "longitude": 10.000654,
                  "zoom": 13,
                },
                "name": "Hamburg",
              },
              "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
              "goods": Array [
                "Washer",
                "Breakfast",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 11,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 53.53434,
                "longitude": 9.998654,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 190,
              "rating": 4.3,
              "title": "The house among olive ",
              "type": "house",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 48.85661,
                  "longitude": 2.351499,
                  "zoom": 13,
                },
                "name": "Paris",
              },
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "goods": Array [
                "Air conditioning",
                "Baby seat",
                "Towels",
                "Washer",
                "Breakfast",
                "Fridge",
                "Laptop friendly workspace",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 87,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 48.86561,
                "longitude": 2.350499,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 266,
              "rating": 2.9,
              "title": "Tile House",
              "type": "hotel",
            },
            Object {
              "bedrooms": 4,
              "city": Object {
                "location": Object {
                  "latitude": 50.846558,
                  "longitude": 4.351697,
                  "zoom": 13,
                },
                "name": "Brussels",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "goods": Array [
                "Washer",
                "Laptop friendly workspace",
                "Breakfast",
                "Air conditioning",
                "Baby seat",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 50.86556,
                "longitude": 4.371697,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 218,
              "rating": 4.6,
              "title": "The house among olive ",
              "type": "apartment",
            },
            Object {
              "bedrooms": 2,
              "city": Object {
                "location": Object {
                  "latitude": 50.938362,
                  "longitude": 6.959974,
                  "zoom": 13,
                },
                "name": "Cologne",
              },
              "description": "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
              "goods": Array [
                "Breakfast",
                "Washer",
                "Baby seat",
                "Laptop friendly workspace",
                "Air conditioning",
              ],
              "host": Object {
                "avatarUrl": "img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 34,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
              ],
              "isFavorite": undefined,
              "isPremium": undefined,
              "location": Object {
                "latitude": 50.945362,
                "longitude": 6.935974,
                "zoom": 16,
              },
              "maxAdults": undefined,
              "previewImage": undefined,
              "price": 485,
              "rating": 2.5,
              "title": "Nice, cozy, warm big bed apartment",
              "type": "house",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
